src/main/resources/lib/unsplash/client.ts

```diff
@@ -0,0 +1,53 @@
+import { request } from "/lib/http-client";
+import { SearchPhotosParams, UnsplashResponseBody } from "/lib/unsplash/types";
+
+export function searchPhotos({
+  query,
+  count,
+  start,
+}: SearchPhotosParams): UnsplashResponseBody {
+  const unsplashPageNumber = Math.floor(start / count);
+
+  const accessKey = app.config.unsplashAccessKey;
+  const url = "https://api.unsplash.com/search/photos";
+  const response = request({
+    url,
+    headers: {
+      Authorization: `Client-ID ${accessKey}`,
+    },
+    params: {
+      query,
+      page: unsplashPageNumber.toString(),
+      per_page: count.toString(),
+    },
+  });
+
+  if (response.status < 400 && response.body) {
+    return JSON.parse(response.body);
+  } else {
+    throwErrorFromBodyString(response.body);
+  }
+}
+
+export function getPhotoById(id: string) {
```

Comment:
Your functions should always have return types. That makes the functions into little LEGOs that can click together.

---

This might be a lint setting that is good to turn on?